UI/SidePanel.cpp

```diff
@@ -3211,7 +3211,9 @@ void SidePanel::DoLayout() {
 
     // system name / droplist
     ul = GG::Pt(GG::X(MaxPlanetDiameter()), GG::Y0);
-    lr = ul + GG::Pt(ClientWidth() - GG::X(MaxPlanetDiameter()), name_height);
+    auto system_name_width = ClientWidth() - GG::X(MaxPlanetDiameter());
```

Comment:
Yes.  Two reasons.

First, there are many integral type that are implicitly cast to int, some signed, some unsigned, some larger and some smaller than int.  If you write `X` there may be an implicit cast and perhaps a bug.  `auto` prevents this bug.  

Second, future proofing.  The type on the right side could change.

`GG` uses `X` to represent absolute locations, relative locations and dimensions.  A bug that I have regularly introduced and occasionally fixed 😮 is to replace one for the other.  A reasonable improvement to the code base would be to create 3 types that are not implicitly converted.  In which case `ClientWidth() - X` would return an XDim which is not convertible to X.  Using X here means that this line would just keep working correctly.

You are right though.  For any given line of code, if the inferred type is not a type only known to the compiler, if you replace the inferred type with a written out type, and you know there are no implicit conversions, then you have the exact same generated code.  The point of `auto` is to off-load all of that checking on the compiler.

Fewer opportunities for mistakes.